### ISSUE TITLE
Questions about using LSTM 

### ISSUE THREAD

**Issue by jazzsaxmafia** [tensorflow/tensorflow#140]
Hi. I am a newbie on TensorFlow, just like most of the others here.
I was trying to use the LSTM modules, and some questions arose.

1. How can I cope with non-fixed length of sequences? For example, the length of longest sentence in each batch is different, but in RNN tutorial, it does like 

for i in range(len(num_steps)):
    # The value of state is updated after processing each batch of words.
    output, state = lstm(words[:, i], state)

....

when this "num_steps" is not set, what kind of value or variable can I use?

2. How can I use mask in LSTM? within a batch, all the sentences have different length, so it is essential to mask the lstm results according to the length of each sentence. I don't think there is masking function inside TensorFlor LSTM module. What options do I have?

Sorry for bothering you, and thank you for making public this great project.

-Taeksoo

**Comment by ebrevdo** [comment]
This is a good question.  Our current suggestion is to pad your inputs to a fixed length, then slice them into reasonable frame count chunks (e.g. 50); and use truncated BPTT.  The PTB tutorial has an example of truncated BPTT, but not of padding.  However this is easy enough to do in python for now.

We are considering other ways to work with dynamic length sequences, but nothing is in the release as of now.  But even with purely dynamic RNNs you'll still probably have to pad if you want to minibatch.

A way of dealing with masking is to add a num_steps length list of [batch_size] weight vectors.  Then perform a distributed multiply this with the loss output, aggregate and make that the new loss.  Set weights past your sequence length for each minibatch entry to zero; these particular gradients won't be backpropagated.

**Comment by vrv** [comment]
Thanks for the question!  These types of questions should be posted to the groups list (github issues are for bugs / installation issues).  https://groups.google.com/a/tensorflow.org/forum/#!forum/discuss
